collector/lib/NetworkStatusNotifier.cpp

```diff
@@ -240,12 +242,19 @@ void NetworkStatusNotifier::RunSingle(IDuplexClientWriter<sensor::NetworkConnect
     const sensor::NetworkConnectionInfoMessage* msg;
     ConnMap new_conn_state, delta_conn;
     AdvertisedEndpointMap new_cep_state;
+    bool enableExternalIPs = config_.EnableExternalIPs();
+
     WITH_TIMER(CollectorStats::net_fetch_state) {
-      conn_tracker_->EnableExternalIPs(config_.EnableExternalIPs());
+      conn_tracker_->EnableExternalIPs(enableExternalIPs);
 
       new_conn_state = conn_tracker_->FetchConnState(true, true);
       if (enable_afterglow_) {
         ConnectionTracker::ComputeDeltaAfterglow(new_conn_state, old_conn_state, delta_conn, time_micros, time_at_last_scrape, afterglow_period_micros_);
+        if (!prevEnableExternalIPs && enableExternalIPs) {
+          ConnectionTracker::CloseNormalizedConnections(&old_conn_state, &delta_conn);
+        } else if (prevEnableExternalIPs && !enableExternalIPs) {
+          conn_tracker_->CloseExternalUnnormalizedConnections(&old_conn_state, &delta_conn);
+        }
```

Comment:
Think this might be a bit clearer?
```suggestion
      if (prevEnableExternalIPs != enableExternalIPs) {
        if (enableExternalIPs) {
          ConnectionTracker::CloseNormalizedConnections(&old_conn_state, &delta_conn);
        } else {
          conn_tracker_->CloseExternalUnnormalizedConnections(&old_conn_state, &delta_conn);
        }
        prevEnableExternalIPs = enableExternalIPs  
      }
```
You should also remove the update happening on line 289.

Reply:
I have largely accepted this change. I couldn't press the "Commit suggestion" button, because I made a change that made this outdated. I made this into a function and moved it to `ConnTracker`. I think one downside of moving this to `ConnTracker` is that it is not as clear what happens to `prevEnableExternalIPs` in when looking at `RunSingle`.